### PR TITLE
SDK-2265 Retrieve Receipt

### PR DESCRIPTION
**Pull request by LuborRod**

### CODE REVIEW

**Review by saurabh-yoti — CHANGES_REQUESTED** [review]

File: tests/DigitalIdentityClientTest.php
[the remark this answers is not in the document]
Comment (mehmet-yoti) [reply tests/DigitalIdentityClientTest.php:1]: added test

File: tests/Identity/DigitalIdentityServiceTest.php
[the remark this answers is not in the document]
Comment (mehmet-yoti) [reply tests/Identity/DigitalIdentityServiceTest.php:1]: added test

File: src/Identity/DigitalIdentityService.php [hunk 1]
```diff
@@ -101,9 +103,67 @@ public function fetchShareSession(string $sessionId): ShareSessionFetched
...
+
+            return $receiptParser->createSuccess($wrappedReceipt, $receiptKey, $this->pemFile);
+        }
+
+        return $receiptParser->createFailure($wrappedReceipt);
+    }
+
+    private function doFetchShareReceipt(string $receiptId): WrappedReceipt
+    {
+        $response = (new RequestBuilder($this->config))
```
[the remark this answers is not in the document]
Comment (mehmet-yoti) [reply src/Identity/DigitalIdentityService.php:132]: updated in previous prs.

File: src/Identity/WrappedReceipt.php [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+    private string $wrappedItemKeyId;
+
+    private string $wrappedKey;
+
+    private ?string $rememberMeId;
+
+    private ?string $parentRememberMeId;
+
+    private ?string $error;
```
[the remark this answers is not in the document]
Comment (mehmet-yoti) [reply src/Identity/WrappedReceipt.php]: updated in previous prs.

File: src/Identity/ReceiptItemKey.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    public function setIv(string $iv): void
+    {
+        $decodedProto = base64_decode($iv, true);
+        if ($decodedProto === false) {
+            throw new EncryptedDataException('Could not decode data');
+        }
+        $encryptedDataProto = new EncryptedData();
+        $encryptedDataProto->mergeFromString($decodedProto);
```
[the remark this answers is not in the document]
Comment (mehmet-yoti) [reply src/Identity/ReceiptItemKey.php]: updated in previous prs.

**Review by saurabh-yoti — APPROVED** [review]

**Review by saurabh-yoti — APPROVED** [review]

**Pull request merged**